extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -1730,6 +1737,8 @@ public enum ResponseMode {
          */
         public Optional<Boolean> userInfoRequired = Optional.empty();
 
+        private AtomicBoolean dynamicallyEnabledUserInfo = null;
```

Comment:
What case does it attempt to support ? We can do it at the build time, but also some other property as far as I recall - like verifying code flow access tokens.

Can we try to avoid having to add some specific code per every such property ?

Reply:
@michalvavrik 
>because you don't know whether OIDC Provider supports user info until you have metadata. You cannot do it until you receive response from OIDC provider (if discovery is enabled).

But how is it relevant to what this PR is about ? The user's intention to require it in properties is not related to whether it can actually be satisfied. Furthemore, with OAuth2 providers, no userInfo endpoint exists.

Reply:
> But how is it relevant to what this PR is about ? The user's intention to require it in properties is not related to whether it can actually be satisfied.

Look here https://github.com/quarkusio/quarkus/blob/b11cdaf9a39f2f1bd4e6ef11a3a5b01d0ad74360/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java#L550. We only require user info if we detect a CDI injection point and provider supports userInfo.

> Furthemore, with OAuth2 providers, no userInfo endpoint exists.

I don't understand this note. Can you clarify why is it relevant? I suspect you are talking about something else than this code does.

Reply:
in other words, the line you are commenting on exists so that I can migrate this line https://github.com/quarkusio/quarkus/blob/b11cdaf9a39f2f1bd4e6ef11a3a5b01d0ad74360/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java#L550.

Reply:
@michalvavrik This bean contains either discovered metadata or those that is manually configured... In case of OAuth2 providers, ex, GitHub or Strava etc, we point manually to some provider specific account info. Or user may have configured it manually for whatever reasons.

I'm sorry, but I'm really not sure what we are trying to achieve with this UserInfo related updates. IMHO it is not exactly relevant to what this PR plans to do.

Reply:
Please have a look again and tell me how is it not relevant:

1. if https://github.com/quarkusio/quarkus/blob/b11cdaf9a39f2f1bd4e6ef11a3a5b01d0ad74360/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java#L550 passes
2. https://github.com/quarkusio/quarkus/blob/b11cdaf9a39f2f1bd4e6ef11a3a5b01d0ad74360/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java#L551 is invoked
3. https://github.com/quarkusio/quarkus/blob/b11cdaf9a39f2f1bd4e6ef11a3a5b01d0ad74360/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantContextFactory.java#L367 user info field can be modified

How can I ignore it? I think there is serious misunderstanding and I am not sure how to explain situation differently.

Reply:
That ^ is current main.

Reply:
@michalvavrik But this is done by the recorder, or as part of the dynamic tenant creation - it is modified like any other field can be. Like I said, you may have a point and we need to tighten it here, let's just have a targeted issue/discussion/PR, you may be right that we are talking about different things, so let's have target each of this issues individually for us to come to the common understanding

Reply:
> it is modified like any other field can be

just for the record, this is the only scenario that needs to be modified after the config is accepted by Quarkus OIDC, every other modification is done once, when Quarkus OIDC gets `OidcTenantConfig` instance from either user or SR Config.